test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -240,13 +248,23 @@ public InternalTestCluster(String nodeMode, long clusterSeed, Path baseDir,
 
         Random random = new Random(clusterSeed);
 
-        this.numSharedAllRolesNodes = RandomInts.randomIntBetween(random, minNumDataNodes, maxNumDataNodes);
-        assert this.numSharedAllRolesNodes >= 0;
+        this.numSharedDataNodes = RandomInts.randomIntBetween(random, minNumDataNodes, maxNumDataNodes);
```

Comment:
I think you meant to comment on the numShardMasterNodes? - I find such a mouthful and I'm not sure it really helps - I opted to have the same naming conversion as numShardDataNodes